pkg/image/api/v1/trigger/trigger.go

```diff
@@ -0,0 +1,24 @@
+package trigger
+
+// ObjectFieldTrigger links a field on the current object to another object for mutation.
+type ObjectFieldTrigger struct {
```

Comment:
So as I was going through this change I realized a few things.

1. Duplicate target fields should not be allowed, because image triggers are level driven (so the latest level has to be stable)
2. If duplicate targets are not allowed, target field paths are unique.  
3. All the new code only uses fieldPath, although I need to switch from index to name to avoid races

If we do that, UID is unnecessary, only target ID.  Since we're level driven, as long as the ordering in the controller is correct, we'll always converge.